fpdf/fpdf_renderer.py

```diff
@@ -0,0 +1,437 @@
+"""
+Based on https://github.com/matplotlib/matplotlib/blob/v3.7.1/lib/matplotlib/backends/backend_template.py
+
+Just need to tell MatPlotLib to use this renderer and then do fig.savefig.
+"""
+
+from contextlib import nullcontext
+import logging
+from matplotlib.backend_bases import (
+    FigureCanvasBase,
+    FigureManagerBase,
+    GraphicsContextBase,
+    RendererBase,
+)
+from matplotlib.transforms import Affine2D
+import matplotlib as mpl
+
+from fpdf.drawing import PaintedPath
+from fpdf.enums import PathPaintRule
+
+PT_TO_MM = 0.3527777778  # 1 point = 0.3527777778 mm
+
+
+_log = logging.getLogger(__name__)
+
+
+class RendererTemplate(RendererBase):
+    """Removed draw_markers, draw_path_collection and draw_quad_mesh - all optional, we can add later"""
```

Comment:
This is not a dosctring describing this class role & behaviour, more of a development comment.